zkabacus-crypto/src/states.rs

```diff
@@ -182,6 +191,11 @@ impl MerchantBalance {
         }
     }
 
+    /// Indicate if the balance is zero.
+    pub fn is_zero(&self) -> bool {
```

Comment:
The `Balance` struct is an internal type that's supposed to reduce code repetition in the public `Merchant/CustomerBalance` types. So yeah, it probably should have an `is_zero()` method, but I'm not sure that a trait would be useful because we never abstract over these types.